core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -341,7 +342,7 @@ object KafkaMetadataLog {
         }
       }
 
-    val replicatedLog = new KafkaMetadataLog(log, snapshotIds, topicPartition, maxFetchSizeInBytes)
+    val replicatedLog = new KafkaMetadataLog(log, scheduler, snapshotIds, topicPartition, maxFetchSizeInBytes)
```

Comment:
Not something we need to solve here, but I think we should put some thought into consolidating file management. Right now it's a little awkward to divide responsibility between `KafkaMetadataLog` and `Log`. For example, I think we are trying to say that `KafkaMetadataLog` is responsible for snapshots. That is mostly true, but we are relying on `Log.loadSegmentFiles` for the deletion of orphaned `.delete` snapshots.

Reply:
I also find it's awkward to represent snapshot files using `OffsetAndEpoch` but we need to rely on `Snapshots` to remove and delete snapshots, will file a ticket to discuss consolidating file management, the first thought is to add  `MetadataLogSnapshot` class to represent a snapshot file.